extension/goext/error.go

```diff
@@ -15,43 +15,127 @@
 
 package goext
 
-import "fmt"
-
-// ErrorCode identifies an error code
-type ErrorCode int
+import (
+	"fmt"
+	"net/http"
+)
 
-// Error represents an error
+// Error represents an error code with related HTTP status
 type Error struct {
-	Code ErrorCode
-	Err  error
+	Err    error
+	Status int
 }
 
 // NewError returns a new error
-func NewError(code ErrorCode, err error) Error {
-	return Error{
-		Code: code,
-		Err:  err,
+func NewError(status int, err error) *Error {
```

Comment:
if err is an goext.Error I think we should extract the error message without the code. It would be helpful when extensions call 
```
if err := TriggerEvent(...); err != nil { 
    return goext.NewError(..., err) 
}
```

Reply:
good idea - fixed. Please take a look at error_test.go how the stacking works now.